app/crud/crud_rule.py

```diff
@@ -0,0 +1,38 @@
+import traceback
+
+from fastapi.encoders import jsonable_encoder
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+
+from crud.base import CRUDBase
+from models import Rule, Condition
+from schemas import CreateRule, UpdateRule
+
+
+class CrudRule(CRUDBase[Rule, CreateRule, UpdateRule]):
+
+    def get_all(self, db: Session):
+        return db.query(Rule).all()
+
+    def create(self, db: Session, obj_in: CreateRule, **kwargs) -> Rule:
```

Comment:
Why you are overwriting method `create()` and you are not using `super().create()` ?

Reply:
Because the schema contains additional information about conditions that isn't used in the create crud, that the default create crud tries to include when dumping from a dict
Because of this I had two ideas:
1. The one implemented, which was an extended create crud that properly saved the model
2. To create two schemas, one for the api that contained all info, and another for the crud that had only the necessary parts. This solution also requires that the data be transferred from one schema to the other when sending to crud, because the generic crud just dumps the whole dict, expecting it to be precisely the expected values, nothing more nor less

I've refactored the code to be like the second solution, and added another line to the api that creates a CreateRule schema and populates it from the other schema
This also means I'll be removing the crud because it is unnecessary